htmresearch/frameworks/nlp/imbu.py

```diff
@@ -127,20 +133,25 @@ def _modelFactory(self, modelType, **kwargs):
     kwargs.update(**self._defaultModelFactoryKwargs())
 
     if modelType == "CioWordFingerprint":
-      kwargs.update(retina=kwargs.get("retina") or self.defaultRetina,
-                    apiKey=kwargs.get("apiKey"),
+      kwargs.update(modelDir=savePath,
+                    retina=self.retina,
+                    apiKey=self.apiKey,
```

Comment:
If `modelDir`, `retina`, and `apiKey` are in all the if blocks, couldn't they just be in one `kwargs` update or included in `self._defaultModelFactoryKwargs()`?